src/lib/core/SingleAgentClient.ts

```diff
@@ -306,10 +323,11 @@ export class SingleAgentClient {
    * ```
    */
   async handleWebhook(
-    payload: WebhookPayload | any,
+    payload: MCPWebhookPayload | A2ATask | TaskStatusUpdateEvent,
+    taskType: string,
+    operationId: string,
     signature?: string,
     timestamp?: string | number,
-    taskType?: string
   ): Promise<boolean> {
     // Verify signature if secret is configured
     if (this.config.webhookSecret) {
```

Comment:
if webhook secret is set, might want to validate the timestamp is recent (like within 5 min) to prevent replay attacks. right now its accepted as is.